numpy/core/tests/test_regression.py

```diff
@@ -2214,6 +2214,20 @@ def test_reshape_size_overflow(self):
             new_shape = (2, 7, 7, 43826197)
         assert_raises(ValueError, a.reshape, new_shape)
 
+    def test_invalid_structured_dtypes(self):
+        # gh-2865
+        # the following invalid dtypes were causing segfaults
+        assert_raises(ValueError, np.dtype, ('O', [('name', 'i8')]))
+        assert_raises(ValueError, np.dtype, ('i8', [('name', 'O')]))
+        # these were causing SystemErrors
+        assert_raises(ValueError, np.dtype,
+                      ('i', {'name': ('i', 0, 'title', 'oops')}))
+        assert_raises(ValueError, np.dtype,
+                      ('i', {'name': ('i', 'wrongtype', 'title')}))
+        # this one is allowed as a special case though
+        a = np.ones(1, dtype=('O', [('name', 'O')]))
```

Comment:
Thanks for the feedback. I noticed that `test_dtype.py` already has a test that requires that this case works:

    def test_base_dtype_with_object_type(self):
        # Issue gh-2798, should not error.
        np.array(['a'], dtype="O").astype(("O", [("name", "O")]))

There is a discussion at #2798, but I don't really understand the rationale. Hmm, I also notice that you can currently have a dtype with object fields in the first element of the tuple, like this:

    >>> np.zeros(1, dtype=('O,O', 'O,i8'))
    array([(0, 0)], 
          dtype=[('f0', 'O'), ('f1', '<i8')])

or even multiple fields in the first element and just `'O'` in the second one, like this:

    >>> np.zeros(1, dtype=('i4,i4', 'O'))
    array([(0, 0)], 
          dtype=[('f0', '<i4'), ('f1', '<i4')])

but this seems especially useless, since one of the two dtypes in the tuple is just ignored (as long as it has the correct size), even if it doesn't make sense to map it to the other one.

So should we allow all of the dtypes that currently work just in case somebody is using them for some reason, or just ones that kind of make sense (e.g. `('O,O', 'O,O')`), or just the one that is already tested for, or none of them?